src/DssSpell.t.base.sol

```diff
@@ -266,22 +266,53 @@ contract DssSpellTestBase is Config, DssTest {
     function setUp() public {
         setValues(address(chief));
 
-        spellValues.deployed_spell_created = spellValues.deployed_spell != address(0) ? spellValues.deployed_spell_created : block.timestamp;
+        if (spellValues.deployed_spell != address(0)) {
+            spell = DssSpell(spellValues.deployed_spell);
+            if (spell.eta() != 0) {
+                // if we have a scheduled spell in the config
+                // we want to roll our fork to the block where it was deployed
+                // this means the test suite will continue to accurately pass/fail
+                // even if mainnet has already scheduled/cast the spell
+                vm.makePersistent(address(this));
+                vm.makePersistent(address(rates));
+                vm.makePersistent(address(addr));
+                vm.makePersistent(address(deployers));
+                _rollToDeployedSpellBlock(address(spell));
+            }
+        } else {
+            spell = new DssSpell();
+        }
+
         _castPreviousSpell();
-        spell = spellValues.deployed_spell != address(0) ?
-            DssSpell(spellValues.deployed_spell) : new DssSpell();
-
-        if (spellValues.deployed_spell_block != 0 && spell.eta() != 0) {
-            // if we have a deployed spell in the config
-            // we want to roll our fork to the block where it was deployed
-            // this means the test suite will continue to accurately pass/fail
-            // even if mainnet has already scheduled/cast the spell
-            vm.makePersistent(address(this));
-            vm.makePersistent(address(rates));
-            vm.makePersistent(address(addr));
-            vm.makePersistent(address(deployers));
-            vm.rollFork(spellValues.deployed_spell_block);
+    }
+
+    function _rollToDeployedSpellBlock(address _spell) internal {
+        if (_spell == address(0)) {
```

Comment:
Not sure I understand this part. The != address(0) check is already happening before calling this function.

Reply:
Reusability.

Reply:
A general version of this could probably be moved into dss-test